web/modules/custom/farm_nfa/js/farmOS.map.behaviors.farm_nfa_gfw_layers.js

```diff
@@ -2,13 +2,83 @@
     farmOS.map.behaviors.farm_nfa_gfw_layers = {
       attach: async function (instance) {
         // Add layers for fire and deforestation alerts in the GFW plan tab
-        farmNfaPlotGfwApiMap(instance,'fire', 'https://data-api.globalforestwatch.org/dataset/nasa_viirs_fire_alerts/v20220726/query/json');
-        farmNfaPlotGfwApiMap(instance,'deforestation', 'https://data-api.globalforestwatch.org/dataset/gfw_integrated_alerts/v20230215/query/json');
+        const fireAlertsUrl = 'https://data-api.globalforestwatch.org/dataset/nasa_viirs_fire_alerts/v20220726/query/json';
+        const deforestationAlertsUrl = 'https://data-api.globalforestwatch.org/dataset/gfw_integrated_alerts/v20230215/query/json';
+        farmNfaPlotGfwApiMap(instance, 'fire', fireAlertsUrl);
+        farmNfaPlotGfwApiMap(instance,'deforestation', deforestationAlertsUrl);
+        const dateApplyButton = document.querySelector('.comiseo-daterangepicker-buttonpanel button');
+        const dateUpdateButtons = document.querySelectorAll('[role="menuitem"]');
+        dateUpdateButtons.forEach((dateUpdateButton) => {
+          dateUpdateButton.addEventListener('click', () => {
+            updateMapLayers(instance, fireAlertsUrl, deforestationAlertsUrl);
+          });
+        });
+        dateApplyButton.addEventListener('click', () => {
+          updateMapLayers(instance, fireAlertsUrl, deforestationAlertsUrl);
+        });
       }
     }
 }())
 
-async function farmNfaPlotGfwApiMap(instance, mapType, gfwApiUrl) {
+async function updateMapLayers(instance, fireAlertsUrl, deforestationAlertsUrl) {
+  const dateRange = await getStartEndDate();
+  const map = instance.map;
+  const layers = map.getLayers().getArray();
+  const noOfLayers = layers.length;
+  for (let i = 2; i < noOfLayers; i++) {
+    await map.removeLayer(layers[layers.length - 1]);    
+  }
+  farmNfaPlotGfwApiMap(instance, 'fire', fireAlertsUrl, dateRange);
+  farmNfaPlotGfwApiMap(instance, 'deforestation', deforestationAlertsUrl, dateRange);
+}
+
+function getStartEndDate() {
```

Comment:
We need way more documentation than this.

Reply:
@pcambra just confirming, documentation as in more comments and explanation here right ?

Reply:
Yes